File: projetopp/ppv1/Projeto PP - APP/src/app/model/promocao.ts
export class Promocao {
  idPromocao: number;
  idUsuario: number;
  idCategoria: number;
  idSupermercado: number;
  nome: string;
  valor: number;
  dataInicio: string;
  dataTermino: string;
  relevancia: number;
  status: string;
  descricao: string;
  foto: string;

  constructor() {
    this.idPromocao = 0;
    this.idUsuario = 0;
    this.idCategoria = 0;
    this.idSupermercado = 0;
    this.nome = "";
    this.valor = 0;
    this.dataInicio = "";
    this.dataTermino = "";
    this.relevancia = 0;
    this.status = "";
    this.descricao = "";
    this.foto = "";
  }
}

File: projetopp/API/src/main/java/br/projetopp/appsuper/model/Promocao.java
package br.projetopp.appsuper.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@ToString

public class Promocao {
    private int idPromocao;
    private int idUsuario;
    private int idCategoria;
    private int idSupermercado;
    private String nome;
    private double valor;
    private String dataInicio;
    private String dataTermino;
    private String relevancia;
    private String status;
    private String descricao;
    private String foto;
}
